### ISSUE TITLE
Corrupts Data For Whole Family When Attempting to Change Label To An Existing Label

### ISSUE THREAD

**Issue by stanparker** [PronghornDigital/Montana-News-Archive#85]
I just attempted to change a record's label to the same name as an existing label, and now no records for that family appear when I search for them, but the family name does.

The console displayed this error:

```
Error: ENOTEMPTY: directory not empty, rename '/var/archives/data/gehl_19' -> '/var/archives/data/gehl_8'
    at Error (native)
```

**Comment by DavidSouther** [comment]
Yes, this is a huge issue and needs to be either blocked from occuring or handled otherwise gracefully.

We've discussed this obliquely before, but what do you expect to happen when you have two records:

Adam A
Adam B

And then rename Adam B to Adam A?

1) Merge B into A, save A, delete B
2) Have two records, Adam A and Adam A (this is a fundamental shift in our data model and integrity)
3) Cancel the operation, and inform the user Adam A already exists?
4) Hybrid of 1, 2, and 3, informing about conflict and asking what resolution to take?

**Comment by stanparker** [comment]
Let's do 3 and resolve the issue.

(In the future, perhaps as an enhancement, I could see it being useful to allow two "Adam A"s as long as they're in different families, but that's not a priority)

**Comment by stanparker** [comment]
This is what's in the error console after. It actually looks like not a hard fix, and as I guess it's not actually corrupting the data.

[The link to error.](https://docs.angularjs.org/error/ngRepeat/dupes?p0=item%20in%20valu%E2%80%A6%7B%22path%22:%22stan_1%2Fmtna-3.mp4%22%7D%5D,%22deleted%22:false%7D)
```
angular.js:12546 Error: [ngRepeat:dupes] http://errors.angularjs.org/1.5.0-beta.1/ngRepeat/dupes?p0=item%20in%20valu…%7B%22path%22%3A%22stan_1%2Fmtna-3.mp4%22%7D%5D%2C%22deleted%22%3Afalse%7D
    at Error (native)
    at http://cdnjs.cloudflare.com/ajax/libs/angular.js/1.5.0-beta.1/angular.min.js:6:423
    at http://cdnjs.cloudflare.com/ajax/libs/angular.js/1.5.0-beta.1/angular.min.js:281:359
    at Object.fn (http://cdnjs.cloudflare.com/ajax/libs/angular.js/1.5.0-beta.1/angular.min.js:130:264)
    at n.$digest (http://cdnjs.cloudflare.com/ajax/libs/angular.js/1.5.0-beta.1/angular.min.js:131:356)
    at n.$apply (http://cdnjs.cloudflare.com/ajax/libs/angular.js/1.5.0-beta.1/angular.min.js:134:374)
    at g (http://cdnjs.cloudflare.com/ajax/libs/angular.js/1.5.0-beta.1/angular.min.js:88:340)
    at H (http://cdnjs.cloudflare.com/ajax/libs/angular.js/1.5.0-beta.1/angular.min.js:92:461)
    at XMLHttpRequest.B.onload (http://cdnjs.cloudflare.com/ajax/libs/angular.js/1.5.0-beta.1/angular.min.js:93:498)(anonymous function) @ angular.js:12546(anonymous function) @ angular.js:9307n.$digest @ angular.js:15913n.$apply @ angular.js:16166g @ angular.js:10607H @ angular.js:10805B.onload @ angular.js:10746
```

**Comment by stanparker** [comment]
It looks like all we have to do is add "$track by index"

**Comment by DavidSouther** [comment]
So, do it ;)

**Comment by stanparker** [comment]
Okay, so I did and I got it to work locally (yay!). But then I realized that's not a good solution because even though it fixes the problem on the front-end,  the data folders are based on the labels, so they really HAVE to be unique.

So I think this is a P0 bug until we totally block the user from creating a duplicate label and tell them why.

**Comment by DavidSouther** [comment]
> the data folders are based on the labels, so they really HAVE to be unique.

Yes, I've made that statement as well :)

> block the user from creating a duplicate label and tell them why.

Ok.

**Comment by stanparker** [comment]
Fixed by #87
